### DISCUSSION TITLE
How to combine spancat and relations

### DISCUSSION THREAD

**Discussion by cliuxinxin** [explosion/spaCy#11710] [Help: Best practices]
Hello, spacy developers.

Thank you for a great job that makes my life so much easier.

I used NER functionality to meet the needs of the company, and I also used NER and rel functionality for my own small project.

And now I've discovered spancat, which I think is a better fit for my um little project than ner. I'm not sure if spancat and rel work together.

Thank you for all your hard work.

**Comment by polm** [comment]
Glad to hear spaCy has been helpful!

The example REL component doesn't work out of the box with spancat, but it should be possible to make it work. You'd need to modify the code to use the spangroups assigned by your spancat instead of entities on the Doc. I don't think it'd be that involved a conversion, but feel free to let us know if there's something we can help with.

**Reply by polm** [reply]
Actually, taking a quick look at it, it looks like the only place you'd need to modify to get it working is [the instance generator](https://github.com/explosion/projects/blob/9cf98235b9d6f4cc3c6bf996d8dc90fe60b8b459/tutorials/rel_component/scripts/rel_model.py#L28). That's designed so that you can register your own alternative generator instead, too, so you can copy it, give it a new name, and modify your config accordingly. 

The evaluation script would also need modification, and technically the component could be changed to not rely on `doc.ents`, but that's more of a bookkeeping detail, and shouldn't affect functionality.

**Comment by cliuxinxin** [comment]
Thank you for your relpying. I will have it a try
